pycloudlib/lxd/instance.py

```diff
@@ -11,20 +12,51 @@ class LXDInstance(BaseInstance):
 
     _type = 'lxd'
 
-    def __init__(self, name):
+    def __init__(self, name, key_pair=None):
         """Set up instance.
 
         Args:
             name: name of instance
+            key_pair: SSH key object
         """
-        super().__init__(key_pair=None)
+        super().__init__(key_pair=key_pair)
 
         self._name = name
+        self._is_vm = None
```

Comment:
This should be a class attr, no need to set in __init__ we can reset that via is_vm() later as you already do
```diff
diff --git a/pycloudlib/lxd/instance.py b/pycloudlib/lxd/instance.py
index 20ddf7d..9594f47 100644
--- a/pycloudlib/lxd/instance.py
+++ b/pycloudlib/lxd/instance.py
@@ -11,6 +11,7 @@ class LXDInstance(BaseInstance):
     """LXD backed instance."""
 
     _type = 'lxd'
+    _is_vm = None
 
     def __init__(self, name, key_pair=None):
         """Set up instance.
@@ -22,7 +23,6 @@ class LXDInstance(BaseInstance):
         super().__init__(key_pair=key_pair)
 
         self._name = name
-        self._is_vm = None
 
     def __repr__(self):
         """Create string representation for class."""
```